File: ml-pipeline/pipelines/extract_skills.js
import { callLLM } from '../integrations/llm.js';
import { loadPrompt } from '../utils/load_prompt.js';

export async function extractSkills(jobDescription) {
  const prompt = loadPrompt('extract_job_skills.md', { job_description: jobDescription });
  const response = await callLLM(prompt, { maxTokens: 2048 });

  // Strip markdown code fences if Claude wrapped the JSON
  const jsonMatch = response.match(/\{[\s\S]*\}/);
  if (!jsonMatch) throw new Error('extract_skills: no JSON found in response');

  return JSON.parse(jsonMatch[0]);
}

File: services/tailored_resume_generation/ml_pipeline/pipelines/extract_skills.py
import json
import logging
import re

from ..integrations.llm import call_llm
from ..utils.load_prompt import load_prompt

logger = logging.getLogger(__name__)


def extract_skills(job_description: str) -> dict:
    prompt = load_prompt("extract_job_skills.md", {"job_description": job_description})
    response = call_llm(prompt, max_tokens=2048)
    logger.debug("[extract_skills] Raw LLM response:\n%s", response)

    match = re.search(r'\{[\s\S]*\}', response)
    if not match:
        logger.error("[extract_skills] No JSON found in response. Full response was:\n%s", response)
        raise ValueError("extract_skills: no JSON found in response")

    return json.loads(match.group(0))
